python/lsst/meas/base/transforms.py

```diff
@@ -0,0 +1,103 @@
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""Measurement transformations.
+
+When a measurement plugin is run, it provides raw, uncalibrated outputs such
+as pixel positions. A transformation may be run as a post-processing step to
+convert those outputs to calibrated quantities, such as celestial coordinates.
+
+At construction, the transformation is passed the configuration and name of
+the plugin whose outputs it will be transformaing (all fields in the input
+table produced by that plugin will have their field names prefixed by the
+plugin name) and a `SchemaMapper` which holds the schemata for the input and
+output catalogs and which may be used to directly map fields between the
+catalogs.
+
+When a transformer is called, it is handed a `SourceCatalog` containing the
+measurements to be transformed, a `BaseCatalog` in which to store results, and
+information about the WCS and calibration of the data. It may be safely
+assumed that both are contiguous in memory, thus a ColumnView may be used for
+efficient processing.
+
+Transformations can be defined in Python or in C++. Python code should inherit
+from `MeasurementTransform`, following its interface. C++ code inherits from
+`BaseTransform`. `TransformWrapper` wraps the C++ implementation so that it
+provides a uniform interface with Python.
+"""
+
+__all__ = ("NullTransform", "PassThroughTransform", "TransformWrapper")
+
+class MeasurementTransform(object):
+    """!
+    Base class for measurement transformations.
+
+    Create transformations by deriving from this class, implementing
+    `__call__()` and (optionally) augmenting `__init__()`.
+    """
+    def __init__(self, config, name, mapper):
+        self.name = name
+        self.config = config
+
+    def __call__(self, inputCatalog, outputCatalog, wcs, calib):
+        raise NotImplementedError()
+
+
+class NullTransform(MeasurementTransform):
+    """!
+    The null transform transfers no data from input to output.
+
+    This is intended as the default for measurements for which no other
+    transformation is specified.
+    """
+    def __call__(self, inputCatalog, outputCatalog, wcs, calib):
+        pass
+
+
+class PassThroughTransform(MeasurementTransform):
+    """
+    Copy all fields named after the measurement plugin from input to output, wihtout transformation.
+    """
+    def __init__(self, config, name, mapper):
+        MeasurementTransform.__init__(self, config, name, mapper)
+        for key, field in mapper.getInputSchema().extract(name + "*").itervalues():
+            mapper.addMapping(key)
+
+    def __call__(self, inputCatalog, outputCatalog, wcs, calib):
+        pass
+
+
+class TransformWrapper(object):
+    """
+    Wrap a C++-style transformation.
+
+    C++ code requires a `Control`, rather than a `Config`, object. This class
+    converts the latter to the former when the transformer is instantiated.
```

Comment:
I actually _did_ mean instantiated here, but the way it's written is pretty confusing, so you're quite right to call me on it. I'll make it clearer.